### PR TITLE
MCR-2316 Value substitution in i18n URIResolver

### PR DESCRIPTION
**Pull request by OPNA2608**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2316).

### CODE REVIEW

**Comment by yagee-de** [comment]
How do you handle values that contain special characters: ",", "[", "]"?

**Comment by bt1cu** [comment]
The syntax should be i18n:{property}:{param1} ... more Parameter with :{paramn}

**Review by bt1cu — APPROVED** [review]

File: mycore-base/src/main/java/org/mycore/common/xml/MCRURIResolver.java [hunk 1]
```diff
@@ -1631,7 +1638,15 @@ public Source resolve(String href, String base) {
 
             final Element i18nElement = new Element("i18n");
             if (!target.contains("*") && !target.contains(",")) {
-                i18nElement.addContent(MCRTranslation.translate(target));
+                String translation;
+                if (target.contains(":")) {
+                    translation = new MessageFormat(
```
Comment (sebhofmann) [inline mycore-base/src/main/java/org/mycore/common/xml/MCRURIResolver.java]: Formating should be handled by MCRTranslation:
```
final int i = target.indexOf(":");
MCRTranslation.translate(target.substring(0, i), (Object[]) target.substring(i+1).split(":"));
```
(not tested)

Comment (OPNA2608) [reply mycore-base/src/main/java/org/mycore/common/xml/MCRURIResolver.java]: Appears to work, thanks.

**Review by sebhofmann — APPROVED** [review]

**Pull request merged**